component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -710,7 +710,7 @@ private void handleTotpVerificationFail(AuthenticationContext context) throws Au
                     break;
             }
         }
-        String username = authenticatedUser.getUserName();
+        String username = authenticatedUser.toFullQualifiedUsername();
```

Comment:
Seems like its better to use the `authenticatedUser` instead of the `username` and use it in the methods. We might need to improve the methods to support this. WDYT?

Reply:
Fixed the method. But we cannot remove username variable from here as it is used in later as well.

Reply:
Removed username and move the logic to get username inside the methods